### PR TITLE
Ack messages for all trade, offer- and dispute messages

### PR DESCRIPTION
**Pull request by ManfredKarrer**
See: https://github.com/bisq-network/bisq-desktop/issues/1585

Depends on: https://github.com/bisq-network/bisq-common/pull/26

### CODE REVIEW

**Comment by ripcurlx** [comment]
utACK

File: src/main/java/bisq/network/p2p/AckMessage.java [hunk 1]
```diff
@@ -93,11 +113,40 @@ private AckMessage(String uid,
...
+                ",\n     sourceUid='" + sourceUid + '\'' +
+                ",\n     sourceId='" + sourceId + '\'' +
+                ",\n     result=" + result +
+                ",\n     errorMessage='" + errorMessage + '\'' +
+                "\n} " + super.toString();
+    }
+
+    @Override
+    public long getTTL() {
+        return TimeUnit.DAYS.toMillis(10);
```
[the remark this answers is not in the document]
Comment (ManfredKarrer) [reply src/main/java/bisq/network/p2p/AckMessage.java]: The trade period can take about 1 week (sepa), so 10 days seems to be reasonable safe to not miss acks. Also the dispute can take sometimes longer if a peer need to wait for bank reply or the like. But agree hard to say whats the best value here....

File: src/main/java/bisq/network/p2p/network/Connection.java [hunk 2]
```diff
@@ -322,6 +334,10 @@ public void removeMessageListener(MessageListener messageListener) {
                     "That might happen because of async behaviour of CopyOnWriteArraySet");
     }
 
+    public void addSupportedCapabilitiesListenerAsWeakReference(SupportedCapabilitiesListener listener) {
```
[the remark this answers is not in the document]
Comment (ManfredKarrer) [reply src/main/java/bisq/network/p2p/network/Connection.java]: Yes, its terrible long ;-). Was just too lazy to think of a good short name....

[the remark this answers is not in the document]
Comment (ManfredKarrer) [reply src/main/java/bisq/network/p2p/network/Connection.java]: addWeakCapabilitiesListener should do it... will rename it...

**Pull request merged**